doc/platform_api/CMIS_Diagnostic_Monitoring_Overview_in_SONiC.md

```diff
@@ -0,0 +1,971 @@
+```
+
+### 2.7 Transceiver VDM sample data
+
+The `TRANSCEIVER_VDM_SAMPLE` table stores the real time VDM data
+
+lane_num: Represents lane number of the field. The lane number is an integer value that ranges from 1 to 8.
+
+```plaintext
+    ;Defines Transceiver VDM sample for a port
+    key                                            = TRANSCEIVER_VDM_SAMPLE|ifname    ; information module VDM sample on port
+    ; field                                        = value
+    esnr_media_input{lane_num}                     = FLOAT                  ; eSNR value in dB for media input
+    esnr_host_input{lane_num}                      = FLOAT                  ; eSNR value in dB for host input
+    pam4_level_transition_media_input{lane_num}    = FLOAT                  ; PAM4 level transition parameter in dB for media input
+    pam4_level_transition_host_input{lane_num}     = FLOAT                  ; PAM4 level transition parameter in dB for host input
+    prefec_ber_min_media_input{lane_num}           = FLOAT                  ; Pre-FEC BER minimum value for media input
+    prefec_ber_max_media_input{lane_num}           = FLOAT                  ; Pre-FEC BER maximum value for media input
+    prefec_ber_avg_media_input{lane_num}           = FLOAT                  ; Pre-FEC BER average value for media input
+    prefec_ber_curr_media_input{lane_num}          = FLOAT                  ; Pre-FEC BER current value for media input
+    prefec_ber_min_host_input{lane_num}            = FLOAT                  ; Pre-FEC BER minimum value for host input
+    prefec_ber_max_host_input{lane_num}            = FLOAT                  ; Pre-FEC BER maximum value for host input
+    prefec_ber_avg_host_input{lane_num}            = FLOAT                  ; Pre-FEC BER average value for host input
+    prefec_ber_curr_host_input{lane_num}           = FLOAT                  ; Pre-FEC BER current value for host input
+    errored_frames_min_media_input{lane_num}       = FLOAT                  ; Errored frames minimum value for media input
+    errored_frames_max_media_input{lane_num}       = FLOAT                  ; Errored frames maximum value for media input
+    errored_frames_avg_media_input{lane_num}       = FLOAT                  ; Errored frames average value for media input
+    errored_frames_curr_media_input{lane_num}      = FLOAT                  ; Errored frames current value for media input
+    errored_frames_min_host_input{lane_num}        = FLOAT                  ; Errored frames minimum value for host input
+    errored_frames_max_host_input{lane_num}        = FLOAT                  ; Errored frames maximum value for host input
+    errored_frames_avg_host_input{lane_num}        = FLOAT                  ; Errored frames average value for host input
+    errored_frames_curr_host_input{lane_num}       = FLOAT                  ; Errored frames current value for host input
+```
+
+### 2.8 Transceiver VDM threshold data
```

Comment:
@mihirpat1 should we have separate table ?
TRANSCEIVER_VDM_HIGH_THRESHOLD
TRANSCEIVER_VDM_HWARN_THRESHOLD
TRANSCEIVER_VDM_LOW_THRESHOLD
TRANSCEIVER_VDM_LWARN_THRESHOLD